metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/models/MerkleTree.scala

```diff
@@ -58,9 +61,124 @@ object MerkleTree {
       // Position of the leaf in the lowest level.
       leafIndex: Int,
       // Number of leaves in the lowest level.
+      // TODO: remove? not used when verifying the proof
```

Comment:
Very elegant way of doing it :+1: 

I don't know if it helps people to see this height or not, but I'd say if we don't need it then we can remove it. In the three examples in our codebase 3 out of 2 have this number, but the explanation in the docstrings works without it as well.

Reply:
It looks like the size is used for sanity checking the sibling path and the index itself.

Reply:
You could also deduce the height from `siblingPath`

Reply:
I think it's fine to do the hash check and forego the sanity checking.